src/main/java/ch/uzh/ifi/hase/soprafs24/rest/webFilter/ManualSecurityFilter.java

```diff
@@ -0,0 +1,52 @@
+package ch.uzh.ifi.hase.soprafs24.rest.webFilter;
+
+import ch.uzh.ifi.hase.soprafs24.entity.User;
+import ch.uzh.ifi.hase.soprafs24.service.AuthService;
+import lombok.AllArgsConstructor;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+@WebFilter
+@Order(1)
+@AllArgsConstructor
+public class ManualSecurityFilter  implements Filter {
+    private final AuthService authService;
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        String requestUri = request.getRequestURI();
+
+        if (
+                requestUri.startsWith("/auth") && "POST".equalsIgnoreCase(request.getMethod())
+                        || requestUri.startsWith("/ws") //TODO: ws endpoint auth?
```

Comment:
For /ws/game/{gameid} I think we need to be able to identify the user. Given that we want only authenticated users to see the game overview, we could also have it authed I think.

Reply:
absolutely, ws is just there at the moment because it failed constntly, since the ws sends heartbeat messages. hence i removed it -> was too annoying. And will need to figure out how the ws handles it